hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/fs/CachingSpaceUsageSource.java

```diff
@@ -64,6 +68,7 @@ public CachingSpaceUsageSource(SpaceUsageCheckParams params) {
 
     refresh = params.getRefresh();
     source = params.getSource();
+    lock = new AutoCloseableReadWriteLock(source.toString());
```

Comment:
`source.getPath()` seems to be a better choice, `toString` is easy to modify, but some subclass do not Implement `getPath`

Reply:
`AbstractSpaceUsageSource.toString()` returns `path`.  `getPath()` is not part of the `SpaceUsageSource` interface and I wouldn't add just for this call.

The name of this lock is not too important here.  It would only be used for logging lock/unlock operations, but those are skipped anyway, since all calls here pass `null, null` instead of real logging method.

Reply:
OK, Thanks for your clarification.